roles/cluster_template/tasks/main.yaml

```diff
@@ -0,0 +1,176 @@
+- name: Create common dirs
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: '0755'
+  loop:
+    - "output/{{ clusterName }}"
+    - "output/{{ clusterName }}/feature/odf"
+    - "output/{{ clusterName }}/feature/odf/externalsecrets"
+    - "output/{{ clusterName }}/feature/odf/storageclasses"
+    - "output/{{ clusterName }}/feature/odf/redhatcop.redhat.io"
+    - "output/{{ clusterName }}/externalsecrets"
+    - "output/{{ clusterName }}/nodenetworkconfigurationpolicies"
+
+#Create dirs only found in external cluster
+- name: Create external cluster dirs
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: '0755'
+  loop:
+    - "output/{{ clusterName }}/certificates"
+    - "output/{{ clusterName }}/issuers"
+    - "output/{{ clusterName }}/issuers/openshift-config"
+    - "output/{{ clusterName }}/issuers/openshift-ingress"
+    - "output/{{ clusterName }}/secretstores"
+    - "output/{{ clusterName }}/secretstores/acct-mgt"
+    - "output/{{ clusterName }}/secretstores/group-sync-operator"
+    - "output/{{ clusterName }}/secretstores/openshift-config"
+    - "output/{{ clusterName }}/secretstores/openshift-ingress"
+    - "output/{{ clusterName }}/secretstores/openshift-logging"
+  when: clusterType == "external"
+
+#Create dirs only found in internal clusters
+- name: Create internal cluster dirs
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: '0755'
+  loop:
+    - "output/{{ clusterName }}/clustersecretstores"
+  when: clusterType == "internal"
+
+#Overlay Kustomization.yaml file
+- template:
```

Comment:
You can probably replace most of these `template` paths with a loop. If you structure the `templates` directory appropriately, then the output path is the same relative path as the template path.

Consider the following example; first we iterate over directories and create those in the target; then we iterate over template files and render them in the appropriate destination:

```
- name: create directories
  file:
    state: directory
    path:  "{{ dest }}"
  vars:
    dest: '{{ "output/%s/%s" % (clusterName, item.path) }}'
  when: >-
    item.state == "directory"
  loop: "{{ query('filetree', 'templates/')}}"
  loop_control:
    label: "{{ item.path }}"

- name: apply templates
  template:
    src:  "{{ src }}"
    dest:  "{{ dest }}"
  vars:
    src: "{{ item.path }}"
    dest: '{{ "output/%s/%s" % (clusterName, item.path) }}'
  when: >-
    item.state == "file"
  loop: "{{ query('filetree', 'templates/')}}"
  loop_control:
    label: "{{ item.path }}"
```

The above won't work as written (the existing structure of the templates directory isn't appropriate), but maybe it gives you some ideas.

This is particularly nice because in order to introduce a new template, you simply have to create it in the appropriate location -- no additional changes are required in the playbooks.